care/facility/api/serializers/facility.py

```diff
@@ -152,6 +152,41 @@ def create(self, validated_data):
         return super().create(validated_data)
 
 
+class FacilityHubSerializer(serializers.ModelSerializer):
+    hub = FacilityBareMinimumSerializer(read_only=True)
+    spoke = FacilityBareMinimumSerializer(read_only=True)
+    hub_id = serializers.UUIDField(write_only=True)
+
+    class Meta:
+        model = FacilityHubSpoke
+        fields = (
+            "external_id",
+            "hub",
+            "hub_id",
+            "spoke",
+            "relationship",
+            "created_date",
+            "modified_date",
+        )
+        read_only_fields = ("external_id", "spoke", "created_date", "modified_date")
+
+    def validate(self, data):
+        hub = Facility.objects.get(external_id=data["hub_id"])
+        data["hub"] = hub
+        del data["hub_id"]
+        data["spoke"] = self.context["facility"]
+
+        if data["hub"].external_id == data["spoke"].external_id:
+            raise serializers.ValidationError("Hub and Spoke cannot be same")
+
+        if FacilityHubSpoke.objects.filter(
+            hub=data["hub"], spoke=data["spoke"]
+        ).exists():
+            raise serializers.ValidationError("Hub and Spoke already exists")
```

Comment:
inheriting from the base model adds soft delete, this condition might fail if the old relation is soft deleted